src/components/TodoList/TodoList.tsx

```diff
@@ -1,44 +1,118 @@
 import React from 'react';
 import './TodoList.scss';
+import classNames from 'classnames';
 
-export const TodoList: React.FC = () => (
-  <div className="TodoList">
-    <h2>Todos:</h2>
-
-    <div className="TodoList__list-container">
-      <ul className="TodoList__list">
-        <li className="TodoList__item TodoList__item--unchecked">
-          <label>
-            <input type="checkbox" readOnly />
-            <p>delectus aut autem</p>
-          </label>
-
-          <button
-            className="
-              TodoList__user-button
-              TodoList__user-button--selected
-              button
-            "
-            type="button"
-          >
-            User&nbsp;#1
-          </button>
-        </li>
-
-        <li className="TodoList__item TodoList__item--checked">
-          <label>
-            <input type="checkbox" checked readOnly />
-            <p>distinctio vitae autem nihil ut molestias quo</p>
-          </label>
-
-          <button
-            className="TodoList__user-button button"
-            type="button"
+type Props = {
+  todos: Todo[];
+  selectUser: (userId: number) => void;
+};
+
+type State = {
+  query: string;
+  filterBy: string;
+};
+
+export class TodoList extends React.Component<Props, State> {
+  state: State = {
+    query: '',
+    filterBy: '',
+  };
+
+  handleChange = (event: React.ChangeEvent<HTMLInputElement | HTMLSelectElement>) => {
+    const { name, value } = event.target;
+
+    this.setState({ [name]: value } as Pick<State, keyof State>);
+  };
+
+  getFilteredTodos = () => {
+    const { todos } = this.props;
+
+    let visibleTasks = todos.filter(todo => (todo.title
+      && todo.title.toLowerCase().includes(this.state.query.toLowerCase())));
```

Comment:
this.state.query.toLowerCase() - to the separate variable not to calculate it on every iteration